### PR TITLE
Tool linter: simple test for test params absent from inputs

### PR DESCRIPTION
**Pull request by bernt-matthias**
The IUC tools directory contains approx. 54 such cases. 

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
This is awesome! Are the IUC hits all legitimate ?

File: lib/galaxy/tool_util/linters/tests.py [hunk 1]
```diff
@@ -26,6 +26,30 @@ def lint_tsts(tool_xml, lint_ctx):
                 has_test = True
                 break
 
+        # really simple test that test parameters are also present in the inputs
+        for param in test.findall("param"):
+            name = param.attrib.get("name", None)
+            if not name:
+                lint_ctx.warn("Found test param tag without a name defined.")
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/linters/tests.py]: Should this be an error?

File: test/unit/tool_util/test_tool_linters.py [hunk 2]
```diff
@@ -174,6 +175,36 @@
...
+        <test>
+        </test>
+    </tests>
+</tool>
+"""
+
+TESTS_PARAM = """
+<tool>
+    <inputs>
+        <param name="existent_test_name"/>
```
Comment (nsoranzo) [inline test/unit/tool_util/test_tool_linters.py]: Maybe test `--argument` here?

```suggestion
        <param argument="--existent-test-name"/>
```

File: lib/galaxy/tool_util/linters/tests.py (hunk 1 above)
Comment (bernt-matthias) [reply lib/galaxy/tool_util/linters/tests.py]: Jep. Changed. Note this should also be covered by the xsd https://github.com/galaxyproject/galaxy/blob/e106fd2f0b1a9a2f37a92e634fb8a5dc3d0fe51a/lib/galaxy/tool_util/xsd/galaxy.xsd#L1476

**Comment by bernt-matthias** [comment]
> This is awesome! Are the IUC hits all legitimate ?

See https://github.com/galaxyproject/tools-iuc/issues/4095 I checked a couple of them and all of them were.

I also thought about matching test params and input params. But this seemed to complicated .. but ideas are welcome.

File: test/unit/tool_util/test_tool_linters.py [hunk 3]
```diff
@@ -273,7 +317,9 @@
     'select option definitions',
     'validator imcompatibilities',
     'outputs collection static elements with format_source',
-    'outputs discover datatsets with tool provided metadata'
+    'outputs discover datatsets with tool provided metadata',
+    'test without expectations',
+    'test param missing from inputs'
```
Comment (nsoranzo) [inline test/unit/tool_util/test_tool_linters.py]:
```suggestion
    'test param missing from inputs',
```

File: test/unit/tool_util/test_tool_linters.py [hunk 4]
```diff
@@ -4,7 +4,8 @@
 from galaxy.tool_util.linters import (
     general,
     inputs,
-    outputs
+    outputs,
+    tests
```
Comment (nsoranzo) [inline test/unit/tool_util/test_tool_linters.py]:
```suggestion
    tests,
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**